Algorithm.Python/Alphas/ContingentClaimsAnalysisDefaultPredictionAlpha.py

```diff
@@ -0,0 +1,266 @@
+        self.SetPortfolioConstruction(EqualWeightingPortfolioConstructionModel())
+        
+        ## Set Execution Model
+        self.SetExecution(ImmediateExecutionModel())
+        
+        ## Set Risk Management Model
+        self.SetRiskManagement(NullRiskManagementModel())
+
+
+    def CoarseSelectionFunction(self, coarse):
+        ## Boolean controls so that our symbol universe is only updated once per month
+        if self.Time.month == self.month:
+            return self.symbols
+        else:
+            self.month = self.Time.month
+
+        ## Sort by dollar volume, lowest to highest
+            sortedByDollarVolume = sorted(coarse, key=lambda x: x.DollarVolume, reverse=True)
+
+        ## Filter for assets with fundamental data
+            filtered = [ x.Symbol for x in sortedByDollarVolume ]
+        
+        ## Return smallest 750 -- idea is that smaller companies are most likely to go bankrupt than blue-chip companies
+            self.symbols = filtered[:750]
+
+            return self.symbols
+    
+    
+    def FineSelectionFunction(self, fine):
+        ## Boolean controls so that our symbol universe is updated only at the beginning of each month
+        if self.Time.month == self.month:
+            return self.symbols
+        else:
+            self.month = self.Time.month
+            ## Select symbols with data necessary for our pricing model
+            fineFilter = sorted(fine, key=lambda x: (x.FinancialStatements.BalanceSheet.TotalAssets.OneMonth > 0) and
+                                                    (x.FinancialStatements.BalanceSheet.TotalAssets.ThreeMonths > 0) and
+                                                    (x.FinancialStatements.BalanceSheet.TotalAssets.SixMonths > 0) and
+                                                    (x.FinancialStatements.BalanceSheet.TotalAssets.TwelveMonths > 0) and
+                                                    (x.FinancialStatements.BalanceSheet.CurrentLiabilities.TwelveMonths > 0) and
+                                                    (x.FinancialStatements.BalanceSheet.InterestPayable.TwelveMonths > 0) and
+                                                    (x.OperationRatios.TotalAssetsGrowth.OneYear > 0) and
+                                                    (x.FinancialStatements.IncomeStatement.GrossDividendPayment.TwelveMonths > 0) and
+                                                    (x.OperationRatios.ROA.OneYear > 0), reverse=True)
+
+            self.symbols = [ x.Symbol for x in fineFilter ]
+
+            return self.symbols
+
+
+class ContingentClaimsAnalysisAlphaModel:
+    
+    def __init__(self, *args, **kwargs):
+        self.symbolDataBySymbol = {}
+        self.month = None
+        self.default_threshold = kwargs['default_threshold'] if 'default_threshold' in kwargs else 0.25
+        self.expiry = None
+        self.start_time = kwargs['start_time'] if 'start_time' in kwargs else None
```

Comment:
If `self.start_time` is `None`, `tau` calculation in line 160 will give us:
```
TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'NoneType'
```
So I think it should default to today:
```python
self.start_time = kwargs['start_time'] if 'start_time' in kwargs else \
    datetime.combine(date.today(), datetime.min.time())
```